include/net/ieee802154_radio.h

```diff
@@ -54,6 +54,56 @@ struct ieee802154_filter {
 /* @endcond */
 };
 
+/** ACK frame configuration types. */
+enum ieee802154_ack_config_type {
+	/** Indicates how radio driver should set Frame Pending bit in ACK
+	 *  responses for Data Requests. If enabled, radio driver should
+	 *  determine whether to set the bit or not based on the information
+	 *  provided with ``set_ack_data`` function. Otherwise, Frame Pending
+	 *  bit should be set to ``1`` (see IEEE Std 802.15.4-2006, 7.2.2.3.1).
+	 */
+	IEEE802154_ACK_CONFIG_AUTO_FRAME_PENDING_BIT,
+};
+
+/** ACK frame configuration data. */
+struct ieee802154_ack_config {
```

Comment:
you could generalize it to ieee802154_config, there are other configuration possibilities than the ack itself (long frame for sub-g, tsch parameters, etc...) .

See how it's been made in ethernet.h only one config struct is being used, no need for a data struct then. All can go in one.

Reply:
Ok, I see. Will look into existing solution in `ethernet.h` and try to propose something similar.